src/gui/components.py

```diff
@@ -108,26 +113,63 @@ def __init__(self, text, font, rect, offset):
     def mouse_hover(self):
         return self.rect.collidepoint(mouse_pos())
 
-    # draw
-    def draw_text(self):
-        text_surf = self.font.render(self.text, False, 'Black')
-        text_rect = text_surf.get_frect(center=self.rect.center)
-        self.display_surface.blit(text_surf, text_rect)
-
     def draw_hover(self):
         if self.mouse_hover():
             self.hover_active = True
             pygame.draw.rect(self.display_surface, 'Black', self.rect, 4, 4)
         else:
             self.hover_active = False
 
+    def draw_content(self):
+        self.display_surface.blit(self.content, self._content_rect)
+
     def draw(self, surface):
         self.display_surface = surface
         pygame.draw.rect(self.display_surface, self.color, self.rect, 0, 4)
-        self.draw_text()
+        self.draw_content()
         self.draw_hover()
 
 
+class Button(AbstractButton):
+    """A button that can contain text."""
+
+    def __init__(self, content: str, rect, font):
+        # Force the user to pass a string as content or else raise an error
+        if not isinstance(content, str):
```

Comment:
> By the way, isinstance isn't really appreciated in programming
Where did you get this idea?

> Personally, I wouldn’t pass content in the abstract parent class. Instead, in the ImageButton, add a new parameter picture, and in TextButton, add a new parameter title or text instead of content.
The reason I did this is because the treatment is pretty much the same, only some details are changed since in the end it's an image that gets displayed when the button is drawn. I decided to put isinstance here to enforce the type for each button

Reply:
Using isinstance often leads to code that requires modification to add new types rather than extension


Actually, since ImageButton will only receive images and TextButton will only receive text, we might as well do:

AbstractButton
init(self)

ImageButton
init(self, image)

TextButton
init(self, text)